alfred-telemetry-solr/build.gradle

```
@@ -0,0 +1,124 @@
+        apply plugin: 'eu.xenit.docker'
+
+
+        compileJava {
+            if ("solr4".equals(solrFlavor)) {
+                sourceCompatibility = '1.8'
+                targetCompatibility = '1.8'
+            } else {
+                sourceCompatibility = '11'
+                targetCompatibility = '11'
+            }
+        }
+
+        configurations {
+            // configuration that holds classes to include in the jar
+            commonClasses
+        }
+
+        dependencies {
+            commonClasses project(':alfred-telemetry-solr:common')
+            configurations.compile.extendsFrom(configurations.commonClasses)
+
+            if ("solr4".equals(solrFlavor)) {
+                implementation "org.apache.solr:solr-core:${solrVersion}"
+                implementation "org.alfresco:alfresco-solrclient:${alfrescoVersion}"
+                implementation "org.alfresco:alfresco-solr4:${alfrescoVersion}:classes@jar"
+                implementation "org.apache.tomcat:tomcat-catalina:7.0.92"
+            }   else if ("solr6".equals(solrFlavor)) {
+                implementation "org.apache.solr:solr-core:${solrVersion}"
+                implementation "org.alfresco:alfresco-search:${assVersion}"
+            }
+        }
+
+        // can this be done better?
+        jar {
+            from {
+               configurations.commonClasses.collect {
+                   include 'eu/xenit/**'
+                   include 'io/micrometer/**'
+                   include 'io/prometheus/**'
+                   include 'io/github/mweirauch/**'
+                   include 'com/codahale/**'
+                   zipTree(it)
+               }
+            }
+        }
+
+        task createDockerFile(type: eu.xenit.gradle.docker.tasks.DockerfileWithCopyTask) {
+            dependsOn(jar)
+            group 'docker'
+
+            from "${solrBaseImage}"
+            if("solr4".equals(solrFlavor)) {
+                smartCopy "${rootProject.projectDir}/integration-tests/src/test/resources/${solrFlavor}/solrconfig.xml", "/opt/alfresco/solr4/workspace-SpacesStore/conf/solrconfig.xml"
+                smartCopy "${rootProject.projectDir}/integration-tests/src/test/resources/${solrFlavor}/solrconfig.xml", "/opt/alfresco/solr4/archive-SpacesStore/conf/solrconfig.xml"
+                smartCopy "${rootProject.projectDir}/integration-tests/src/test/resources/${solrFlavor}/95-init-solr-micrometer-metrics.sh", "/docker-entrypoint.d/"
+                smartCopy files(jar).getSingleFile(), "/opt/alfresco/solr4/lib/alfred-telemetry-solr.jar"
+            } else if("solr6".equals(solrFlavor)){
+                dependsOn(configurations.metricsGraphite)
+                // overwrites existing metrics-graphite library with the one used by micrometer
+                smartCopy configurations.metricsGraphite.fileCollection { it.name == "metrics-graphite"}, "/opt/alfresco-search-services/solr/server/lib/"
```

Can you also match the group ID of this dependency? `metrics-graphite` might be the name of multiple dependencies at some point, which would then all be added.

If this library is a dependency of `io.micrometer:micrometer-registry-graphite:1.6.1` (direct or indirect), you can also do

```suggestion
                smartCopy configurations.implementation.fileCollection { it.group == "io.dropwizard.metrics" && it.name == "metrics-graphite"}, "/opt/alfresco-search-services/solr/server/lib/"
```